google/cloud/aiplatform/vizier/study.py

```diff
@@ -0,0 +1,213 @@
+                'learning_rate', 0.00001, 1.0, scale_type=pyvizier.ScaleType.LINEAR)
+            root.add_categorical_param('optimizer', ['adagrad', 'adam', 'experimental'])
+            study = aiplatform.Study.create_or_load(display_name='display_name', problem=sc)
+
+        Args:
+            display_name (str):
+                A name to describe the Study.
+            problem (vz.ProblemStatement):
+                Configurations of the study. It defines the problem to create the study.
+        """
+        api_client = cls._instantiate_client(
+            location=initializer.global_config.location,
+            credentials=initializer.global_config.credentials,
+        )
+        study = gca_study.Study(
+            display_name=display_name, study_spec=problem.to_proto()
+        )
+
+        try:
+            study = api_client.create_study(
+                parent=initializer.global_config.common_location_path(
+                    initializer.global_config.project,
+                    initializer.global_config.location,
+                ),
+                study=study,
+            )
+        except exceptions.AlreadyExists:
+            _LOGGER.info("The study is aleady created. Using existing study.")
+            study = api_client.lookup_study(
+                request={
+                    "parent": initializer.global_config.common_location_path(
+                        initializer.global_config.project,
+                        initializer.global_config.location,
+                    ),
+                    "display_name": display_name,
+                }
+            )
+
+        return Study(study.name)
+
+    def get_trial(self, uid: int) -> TrialInterface:
+        """Retrieves the trial under the study by given trial id."""
+        study_path_components = self._parse_resource_name(self.resource_name)
+        _LOGGER.info(study_path_components)
+        return Trial(
+            Trial._format_resource_name(
+                project=study_path_components["project"],
+                location=study_path_components["location"],
+                study=study_path_components["study"],
+                trial=uid,
+            )
+        )
+
+    def trials(
+        self, trial_filter: Optional[vz.TrialFilter] = None
+    ) -> Collection[TrialInterface]:
+        """Fetches a collection of trials."""
+        list_trials_request = {"parent": self.resource_name}
+        trials_response = self.api_client.list_trials(request=list_trials_request)
+        return [Trial(trial.name) for trial in trials_response.trials]
+
+    def optimal_trials(self) -> Collection[TrialInterface]:
+        """Returns optimal trial(s)."""
+        list_optimal_trials_request = {"parent": self.resource_name}
+        optimal_trials_response = self.api_client.list_optimal_trials(
+            request=list_optimal_trials_request
+        )
+        return [Trial(trial.name) for trial in optimal_trials_response.optimal_trials]
```

Comment:
There is a `_construct_sdk_resource_from_gapic` classmethod that can be used to avoid the extra GET call in the constructor of `Trial`.